File: Code/C++/Factorial_without_recursion.cpp
/*
Factorial of a non-negative integer, is multiplication of all integers smaller than or equal to n.
*/
#include<iostream>
using namespace std;
int main()
{
    int num, fact=1;
    cout<<"\nEnter the number whose factorial you want to find:-";
    cin>>num;
    for(int i=num; i>1; i--) 
    {
        fact=fact*i;
    }
    cout<<"Factorial of "<<num<<" is "<<fact;
    return 0;
}
/*
Test case 1:
	Input : 5
	Output : 120
Test case 1:
	Input : 3
	Output : 6
Time complexity: O(n)
Space Complexity: 1
*/

File: Code/Python/factorial_without_recursion.py
'''
PROBLEM STATEMENT:
Given a number, the task is to print its factorial.
A factorial of a number 'n' is the product of numbers from 1 to n.
'''

# A function to calculate factorial of a number and to print it
def Factorial(n):
    fact = 1
    while n > 0:
        fact = fact*n
        n -= 1
    print(fact)

# Driver Code
n = int(input("Enter a number: "))
print("Factorial of", n, "is:", end=" ")
# A function call to print factorial
Factorial(n)

'''
TEST CASES:
1.
Input:
Enter a number: 5
Output:
Factorial of 5 is: 120

2.
Input:
Enter a number: 8
Output:
Factorial of 8 is: 40320

TIME COMPLEXITY: O(n), for traversing the numbers from 1 to n
where, 'n' denotes the number entered by the user.
SPACE COMPLEXITY: O(1), no extra space used.
'''
